src/coreclr/jit/emitxarch.cpp

```diff
@@ -5927,8 +5933,9 @@ bool emitter::HasSideEffect(instruction ins, emitAttr size)
         case INS_movupd:
         case INS_movups:
         {
-            // non EA_32BYTE moves clear the upper bits under VEX encoding
-            hasSideEffect = UseVEXEncoding() && (size != EA_32BYTE);
+            // TODO-XArch-AVX512 : Handle merge/masks scenarios once k-mask support is added for these.
+            // non EA_32BYTE and EA_64BYTE moves clear the upper bits under VEX and EVEX encoding respectively.
+            hasSideEffect = (UseVEXEncoding() && (size <= EA_32BYTE)) || (UseEvexEncoding() && (size <= EA_64BYTE));
```

Comment:
Shouldn't this be:
```suggestion
if (UseVexEncoding())
{
    if (UseEvexEncoding())
    {
        hasSideEffect = (size < EA_64BYTE);
    }
    else
    {
        hasSideEFfect = (size < EA_32BYTE);
    }
}
else
{
    hasSideEffect = false;
}
```

That is, `UseVexEncoding()` must be true if `UseEvexEncoding` is true (we can support just VEX, but we cannot support just EVEX).

We then have the consideration that:
* Legacy Encoding (no VEX) - preserves upper bits, so no side effect
* VEX - zeros upper bits, so side effect for 16-byte always and side-effect for 32-byte if EVEX is also supported
* EVEX - zeros upper bits, unless masking is specified and mask is "known", so side effecting for 16-byte and 32-byte always.

Reply:
Changed. The `<=` condition was incorrect in my code. Thanks for pointing it out